### PR TITLE
Add hwmon and input interface

### PR DESCRIPTION
**Pull request by jackpot51**
These interfaces will only be active if the latest liberate firmware is utilized. Please test to ensure no regressions on older firmware, and test for sensors and OLED screen toggle support on the liberate firmware built from https://github.com/system76/firmware-open/pull/118

The liberate firmware also integrates these PRs:
- https://github.com/system76/coreboot/pull/32
- https://github.com/system76/ec/pull/93

Rebase and tag when approved

### CODE REVIEW

File: system76_acpi.c [hunk 1]
```diff
@@ -292,6 +454,9 @@ static void system76_notify(struct acpi_device *acpi_dev, u32 event)
 	case 0x84:
 		kb_led_hotkey_color(data);
 		break;
+	case 0x85:
+		input_key(data, KEY_SCREENLOCK);
```
Comment (crawfxrd) [inline system76_acpi.c:458]: Is `KEY_SCREENLOCK` right? I would have assumed `KEY_DISPLAYTOGGLE`.

Comment (jackpot51) [reply system76_acpi.c:458]: KEY_SCREENLOCK is used to lock the screen on the addw2, effectively implementing the screen off key (Fn-F2).
Fn-F7 is already implemented by sending Super+P, which is what the proprietary EC does.

**Review by crawfxrd — APPROVED** [review]

**Review by leviport — APPROVED** [review]

**Comment by d4v3y0rk** [comment]
Does this make the fans information available to sensors?

**Comment by jackpot51** [comment]
It will make it available for the Lemur Pro (lemp9) and Oryx Pro (oryp6) after a firmware update

**Comment by d4v3y0rk** [comment]
@jackpot51 Awesome! Thanks so much for getting us this info! Will it allow us to control the fans or just get info about them?

**Comment by shlevy** [comment]
@jackpot51 I see that this PR is not yet on Linus's tree but is included in the latest release. Is it recommended that users on newer kernels build out of tree or just use their in-kernel version of the module?

**Comment by jackpot51** [comment]
If you want this feature, you will need to use the DKMS module. Upstream Linux has not accepted the change yet.

**Pull request merged**